### PR TITLE
internal/website: add automatic table of contents

### PR DESCRIPTION
**Pull request by zombiezen**
This is enabled per-page using `toc: true` in the front matter. It will float a sidebar to the right on desktop and show the list inline with content after the title on tablet and smaller screens.

### CODE REVIEW

**Comment by zombiezen** [comment]
Screenshot of this in action:

Screenshot of Secrets page with TOC sidebar

**Comment by vangent** [comment]
@zombiezen is there documentation on how to run `hugo` locally to preview the site? @polinasok was looking for that as well.

File: internal/website/static/css/style.css [hunk 1]
```diff
@@ -126,6 +125,39 @@ body {
...
+    border: none;
+    float: none;
+    margin: 1.5rem 0 0.5rem;
+    max-width: none;
+    min-width: none;
+    padding: 0;
+  }
+}
+.PageTOC-heading {
+  font: 700 1rem/1.25 'Work Sans', 'Roboto', sans-serif;
```
Comment (vangent) [inline internal/website/static/css/style.css]: WDYT about making the font smaller? Maybe everywhere (not just TOC), but that's a separate question :-). At least in the screenshot you gave, the TOC is too long to fit on a page and wraps strangely, the font looks too big for me.

**Comment by zombiezen** [comment]
> @zombiezen is there documentation on how to run `hugo` locally to preview the site? @polinasok was looking for that as well.

https://github.com/google/go-cloud/blob/master/internal/website/README.md#local-development

File: internal/website/static/css/style.css (hunk 1 above)
Comment (zombiezen) [reply internal/website/static/css/style.css]: At any legible font size, the page TOC is going to spill off the page for any of our larger (read as: opening) how-to guides, especially on smaller screens. I put in a fair amount of care to make sure the layout doesn't break when this happens.

If you want the font smaller, I'd like to do that as a separate global change. I think some of the issue is coming from the fact that the top header pushes content below the fold.

**Comment by vangent** [comment]
> @vangent
> 
> > @zombiezen is there documentation on how to run `hugo` locally to preview the site? @polinasok was looking for that as well.
> 
> [/internal/website/README.md@`master`#local-development](https://github.com/google/go-cloud/blob/master/internal/website/README.md#local-development) [](/google/go-cloud/blob/HEAD@{2019-07-10T18:13:30Z}/internal/website/README.md#local-development)

Excellent, thanks!

File: internal/website/static/css/style.css (hunk 1 above)
Comment (zombiezen) [reply internal/website/static/css/style.css]: Opened #2504 for the font size change.

**Comment by vangent** [comment]
> Excellent, thanks!

Hmm, I pulled this PR and ran `hugo server -D`, but I don't see any TOCs on the pages...?

**Comment by vangent** [comment]
> > Excellent, thanks!
> 
> Hmm, I pulled this PR and ran `hugo server -D`, but I don't see any TOCs on the pages...?

Oh never mind, I wasn't looking at the right pages.

**Comment by zombiezen** [comment]
@vangent Can you confirm that `git rev-parse HEAD` is 3b4c96fbf229e7487fec0800b13ee93682bf687e? And it's not on all pages, just how-to leaf pages (like `/howto/pubsub/subscribe/`).

File: internal/website/static/css/style.css (hunk 1 above)
Comment (polinasok) [reply internal/website/static/css/style.css]: + 1 to size change for the content. Would TOC size need to be a separate change as well? Seems like it is visually independent of content and can be done here. Also, I think it is important to preserve the nesting structure of the headers.

Comment (zombiezen) [reply internal/website/static/css/style.css]: Done.

**Comment by zombiezen** [comment]
New screenshot (also with a slightly larger window):

Screenshot of secrets how-to guide after changes

**Comment by eliben** [comment]
How does it look on mobile?

**Comment by zombiezen** [comment]
Mobile:

Screenshot of top of secrets page on mobile

**Pull request merged**